docs: review (2023-04) (#12)
schubergphilis/terraform-aws-mcaf-securityhub-findings-manager: README.md

> Previously called: Security Hub Findings Suppressor |
> Based on: <https://github.com/schubergphilis/aws-security-hub-suppressor>

Security Hub Findings Manager is a framework designed to suppress AWS Security Hub using a configurable suppression list.

> This suppression is useful when controls or rules are not applicable in an account. For example, you might want to suppress all DynamoDB Autoscaling configuration findings related to the control `DynamoDB.1` simply because this feature is not applicable for your workload. Besides the suppressing the findings, this module is also able to create Jira tickets for all `NEW` findings with a severity higher than the defined threshold.

Reviewer: change 

```
Besides to suppressing findings
```
to 
```
Besides suppressing findings
```

in the suggestion.

    * [Normalized](https://docs.aws.amazon.com/securityhub/1.0/APIReference/API_Severity.html) severity levels:
      * 0 - `INFORMATIONAL`
      * 1–39 - `LOW`
      * 40–69 - `MEDIUM`
      * 70–89 - `HIGH`
      * 90–100 - `CRITICAL`
* (optional) Step Function, to orchestrate the Suppressor and Jira lambdas.
* YML Configuration File (`suppressor.yaml`) that contains the list of products and the field mapping

## Deployment moded

Each mode deploys a Lambda function that is triggered by upserts to the DynamoDB table and an EventBridge rule which detects new Security Hub findings. In addition to these, additional resources are deployed depending on the chosen deployment mode.

### (Default) without Jira & ServiceNow integration

* The module deploys one Lambda function: `Suppressor`, and configures it as a target to the EventBridge rule watching for new Security Hub findings

### With Jira integration

* The module deploys an additional Lambda function: `Jira`, along with a Step function which orchestrates these Lambda functions and Step Function as a target to the EventBridge rule

> * If the finding is not suppressed and matches the configured severity level, an issue is created in Jira; the Security Hub workflow status will also be changed from `NEW` to `NOTIFIED`

Reviewer 2: ```suggestion
* If the finding is not suppressed and matches the configured severity level, an issue is created in Jira; the Security Hub workflow status will also be changed from `NEW` to `NOTIFIED`

This deployment method can be used by setting the value of the variable `jira_integration` to `true` (default = false).
```

![Step Function Graph](files/step-function-artifacts/securityhub-suppressor-orchestrator-graph.png)

### With ServiceNow integration

[Reference design](https://aws.amazon.com/blogs/security/how-to-set-up-two-way-integration-between-aws-security-hub-and-servicenow)

* The module will deploy additional resources to support ServiceNow integration including (but not limited to): SQS queue, EventBridge rule and the necessary IAM user
* When an event in SecurityHub fires, an event will be created by EventBridge and dropped onto an SQS queue.

> * ServiceNow will pull the events from the SQS queue with the `SCSyncUser` using `acccess_key` & `secret_access_key`.

Reviewer 2: ```suggestion
* ServiceNow will pull the events from the SQS queue with the `SCSyncUser` using `acccess_key` & `secret_access_key`.

This deployment method can be used by setting the value of the variable `servicenow_integration` to `true` (default = false).

```

> **Note**

> > The user will be created by the module, but the `acccess_key` & `secret_access_key` need to be generated in the AWS Console, to prevent storing this data in the Terraform state. If you want Terraform to create the `acccess_key` & `secret_access_key` (and output them), set variable `create_servicenow_access_keys` to `true` (default = false)

Author: ```suggestion
> The user will be created by the module, but the `access_key` & `secret_access_key` need to be generated in the AWS Console, to prevent storing this data in the Terraform state. If you want Terraform to create the `access_key` & `secret_access_key` (and output them), set `var.create_servicenow_access_keys` to `true`.
```

## How it works

> The Security Hub Findings Manager listens to the AWS EventBridge event bus and triggers an execution when the `Security Hub Findings - Imported` event occurs.

Author: ```suggestion
The Security Hub Findings Manager listens to the AWS EventBridge event bus and triggers an execution when the ["Security Hub Findings - Imported event](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-event-formats.html#securityhub-cwe-event-formats-findings-imported) occurs.
```

> Once the event is delivered, the `securityhub-events-suppressor` function will be triggered and will perform the following:

Author: ```suggestion
Once the event is delivered, the `securityhub-events-suppressor` function will perform the following:
```

> 1. Parse the event to determine what is the linked product. For example: Firewall Manager, Inspector or Security Hub.

Reviewer 2: 1 2 3 4 5?
